content/blog/mellanox-switch-automation-welcome-to-2012.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Mellanox switch automation welcome to 2012!
+date: 2022-10-07T22:54:50.118Z
+author: Rick Kauffman
+authorimage: /img/me-art.jpg
+---
+T﻿he Mellanox 2010 ethernet switch is a curious thing. It is only half the width of a normal 19 inch rack mount switch. It has 18 10/25G ports. Perfect when you need two switches for high availability deployments but only a handful of ports are required. The fact they are only half width, you only need 1U of  a data center rack. I believe because of its features and price, it makes an excellent candidate for HPE dHCI deployments.
+
+I﻿ recently was involved with a proof of concept for a very large customer where these Mellanox switches would be used. The end design calls for over 200 locations.A quick calculation tells us that 400 different switches will need to be configured. That would be a lot of typing on a terminal using the switch command line. I think this calls for some automation!
+
+My role in the project is to configure the network to support all the different connections the deployment will require. I should get out my console cable and start hammering away on the command line. After all there are only two configurations at this site, should be easy. 
+
+Fortunately, I love to automate things and I want to have consistency in all my switch configuration files. Let's take a look at the architecture.
+
+ 
+
+![](/img/network_760x435.png "network")
+
+There are only five networks. The ILO or Out of Band management, the VMware Management network, home of vcenter, a VM production network, and two ISCSI data networks.
+
+Looking at a switch configuration file, most of the information is static. Very little of it is dynamic. There are only 17 variables that need to be collected before they can be applied to a Jinja2 template. The solution for me was a form.
```

Comment:
This is the first time you bring up "Jinja2" and the template. Remember that the reader is not in front of you watching what you are doing... you'll need to explain to them somewhere before this that, in part of your automation process, you are going to be using Jinja2 and explain what that is.